core/src/impl/Kokkos_Atomic_Generic.hpp

```diff
@@ -152,7 +152,7 @@ struct RShiftOper {
 
 template < class Oper, typename T >
 KOKKOS_INLINE_FUNCTION
-T atomic_fetch_oper( const Oper& op, volatile T * const dest ,
+T atomic_fetch_oper( const Oper& /* op */, volatile T * const dest ,
```

Comment:
These probably should use op.apply instead of Oper::apply further down, and thus make the parameter used.